internal/agent/collector.go

```diff
@@ -21,113 +20,112 @@ import (
 	libmodel "github.com/konveyor/forklift-controller/pkg/lib/inventory/model"
 	apiplanner "github.com/kubev2v/migration-planner/api/v1alpha1"
 	"github.com/kubev2v/migration-planner/internal/util"
+	"github.com/kubev2v/migration-planner/pkg/log"
 	core "k8s.io/api/core/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type VCenterCreds struct {
-	Url      string `json:"url"`
-	Username string `json:"username"`
-	Password string `json:"password"`
+type Collector struct {
+	log     *log.PrefixLogger
+	dataDir string
 }
 
-func main() {
-	logger := log.New(os.Stdout, "*********** Collector: ", log.Ldate|log.Ltime|log.Lshortfile)
-	// Parse command-line arguments
-	if len(os.Args) < 3 {
-		fmt.Println("Usage: collector <creds_file> <inv_file>")
-		os.Exit(1)
+func NewCollector(log *log.PrefixLogger, dataDir string) *Collector {
+	return &Collector{
+		log:     log,
+		dataDir: dataDir,
 	}
-	credsFile := os.Args[1]
-	outputFile := os.Args[2]
+}
+
+func (c *Collector) collect() {
+	credentialsFilePath := filepath.Join(c.dataDir, CredentialsFile)
 
-	logger.Println("Wait for credentials")
-	waitForFile(credsFile)
+	c.log.Infof("Wait for credentials")
+	waitForFile(credentialsFilePath)
```

Comment:
IMO, now that we've integrated the collector into the agent, maybe it's better to pass the credentials as struct to collector and start the collector only when we get the credentials.
wdytN